src/main/java/billgates/use_cases/user_join/UserJoinUseCase.java

```diff
@@ -36,7 +36,8 @@ public UserJoinUseCase(DatabaseGateway gateway, UserJoinOutputPort outputPort) {
     @Override
     public void join(UserJoinRequestModel model) {
         List<QueryUserData> usersData = this.gateway.getUserData();
```

Comment:
Actually I just realized, we can just test if `this.gateway.getUserData(username)` returns `null` or not. 

Remember we made this method return `null` whenever we cannot find the specified username?